automation.md

```diff
@@ -0,0 +1,102 @@
+
+- Chrome devtool protocol(CDP)
+
+Many tools use CDP, see [awesome-chrome-devtools](https://github.com/ChromeDevTools/awesome-chrome-devtools).
+Related to web automation testing are Puppeteer, Playwright.
+The following is the playwright python code to start the Chrome browser:
+```python
+browser = p.chromium.launch(channel="chrome", headless=False)
+context = browser.new_context()
+# Open new page
+page = context.new_page()
+# Go to https://www.google.com/
+page.goto("https://www.google.com/")
+```
+Using ProcessExplorer, you can see that the chrome start parameters are as follows,
+```
+"C:\Program Files\Google\Chrome\Application\chrome.exe" --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,AcceptCHFrame,AutoExpandDetailsElement --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --no-sandbox --user-data-dir=C:\Users\test\AppData\Local\Temp\playwright_chromiumdev_profile-V3LZUg **--remote-debugging-pipe** --no-startup-window
+```
+According to the chrome startup parameter description:
+```
+--remote-debugging-pipe: Enables remote debug over stdio pipes [in=3, out=4]. Optionally, specifies the format for the protocol messages, can be either "JSON" (the default) or "CBOR"
+```
+
+How to use CDP, you can get started from [this link](https://github.com/aslushnikov/getting-started-with-cdp/blob/master/README.md).
+More crhome startup parameters can be found in the official documentation, or [here](https://github.com/GoogleChrome/chrome-launcher/blob/master/docs/chrome-flags-for-tools.md).
+- Native messaging
+Extensions and apps can exchange messages with native applications using an API that is similar to the other [message passing APIs](https://developer.chrome.com/docs/apps/nativeMessaging/messaging). Native applications that support this feature must register a native messaging host that knows how to communicate with the extension. Chrome starts the host in a separate process and communicates with it using standard input and standard output streams.
+
+More about Native Messaging, please visit [this link](https://developer.chrome.com/docs/apps/nativeMessaging/messaging).
+
+#  SAP Automation
+SAP itself provides script automation, so most RPA products automate SAP applications based on the automation technology provided by SAP itself.
```

Comment:
As SAP provides the script automation by itself, most RPA products automate SAP applications based on its own automation technology.